core/services/relay/evm/config_tracker.go

```diff
@@ -0,0 +1,232 @@
+	ctx, cancel := context.WithCancel(context.Background())
+	return &ConfigTracker{
+		client:              client,
+		addr:                addr,
+		contractABI:         contractABI,
+		chainType:           chainType,
+		latestBlockHeight:   -1,
+		latestBlockHeightMu: sync.RWMutex{},
+		lggr:                lggr,
+		headBroadcaster:     headBroadcaster,
+		unsubscribeHeads:    nil,
+		ctx:                 ctx,
+		ctxCancel:           cancel,
+	}
+}
+
+// Start starts the config tracker in particular subscribing to the head broadcaster.
+func (c *ConfigTracker) Start() error {
+	return c.StartOnce("ConfigTracker", func() (err error) {
+		var latestHead *evmtypes.Head
+		latestHead, c.unsubscribeHeads = c.headBroadcaster.Subscribe(c)
+		if latestHead != nil {
+			c.setLatestBlockHeight(*latestHead)
+		}
+
+		return nil
+	})
+}
+
+// Close cancels and requests and unsubscribes from the head broadcaster
+func (c *ConfigTracker) Close() error {
+	c.ctxCancel()
+	c.unsubscribeHeads()
+	return nil
+}
+
+// Notify not implemented
+func (c *ConfigTracker) Notify() <-chan struct{} {
+	return nil
+}
+
+func callContract(ctx context.Context, addr common.Address, contractABI abi.ABI, method string, args []interface{}, caller contractReader) ([]interface{}, error) {
+	input, err := contractABI.Pack(method, args...)
+	if err != nil {
+		return nil, err
+	}
+	output, err := caller.CallContract(ctx, ethereum.CallMsg{To: &addr, Data: input}, nil)
+	if err != nil {
+		return nil, err
+	}
+	return contractABI.Unpack(method, output)
+}
+
+// LatestConfigDetails queries an OCR2Abstract contract for the latest config details
+func (c *ConfigTracker) LatestConfigDetails(ctx context.Context) (changedInBlock uint64, configDigest ocrtypes.ConfigDigest, err error) {
+	latestConfigDetails, err := callContract(ctx, c.addr, c.contractABI, "latestConfigDetails", nil, c.client)
+	if err != nil {
+		return 0, ocrtypes.ConfigDigest{}, err
+	}
+	// Panic on these conversions erroring, would mean a broken contract.
+	changedInBlock = uint64(*abi.ConvertType(latestConfigDetails[1], new(uint32)).(*uint32))
+	configDigest = *abi.ConvertType(latestConfigDetails[2], new([32]byte)).(*[32]byte)
+	return
+}
+
+// LatestConfig queries an OCR2Abstract contract for the latest config contents.
+func (c *ConfigTracker) LatestConfig(ctx context.Context, changedInBlock uint64) (ocrtypes.ContractConfig, error) {
+	topics, err := abi.MakeTopics([]interface{}{c.contractABI.Events["ConfigSet"].ID})
+	if err != nil {
+		return ocrtypes.ContractConfig{}, err
+	}
+	query := ethereum.FilterQuery{
+		Addresses: []common.Address{c.addr},
+		Topics:    topics,
+		FromBlock: new(big.Int).SetUint64(changedInBlock),
+		ToBlock:   new(big.Int).SetUint64(changedInBlock),
+	}
+	logs, err := c.client.FilterLogs(ctx, query)
+	if err != nil {
+		return ocrtypes.ContractConfig{}, err
+	}
+	if len(logs) == 0 {
+		c.lggr.Warnw("Contract not configured yet")
+		return ocrtypes.ContractConfig{}, err
```

Comment:
Should this be returning an error too? e.g.
```suggestion
		err := errors.New("contract not configured yet")
		c.lggr.Warnw(err)
		return ocrtypes.ContractConfig{}, err
```